### PR TITLE
Components: add validation to the required text input fields in SiteTitle.

### PR DESCRIPTION
**Pull request by AnnaMag**
This patch contains 2 changes:
- adds a prop that can set the blogname field in the the SiteTitle component as required. By default it is false and does not change current functionality. When passed to the component set to `true`, it uses the Calypso style validation to require input for the blogname,
- modifies the site-title step of the JPO to use the above prop and disables the wizard button when no content has been provided.

Context: the Site Title ( blogname) is required in JPO, thus this change.
Addresses part of #21685.

**To test:**
- checkout this branch on your local Calypso
- use the redirect from your JP sandbox `https://YourJetpackSandbox.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development` to land in the first screen of the JPO flow
- verify that empty/filled in site title disables/enables the button and relevant validations are shown

screen shot 2018-01-24 at 00 12 27
screen shot 2018-01-24 at 00 12 12

### CODE REVIEW

File: client/components/site-title/index.jsx [hunk 1]
```diff
@@ -21,13 +22,15 @@ class SiteTitleControl extends React.Component {
 		blogname: PropTypes.string,
 		blogdescription: PropTypes.string,
 		disabled: PropTypes.bool,
+		isTitleRequired: PropTypes.bool,
```
Comment (ockham) [inline client/components/site-title/index.jsx]: This prop name is inconsistent with `autoFocusBlogname` (and arguably with `blogname`). I wouldn't say `blogname` is necessarily better, but props that refer to the same thing should use the same term IMO...

File: client/components/site-title/index.jsx [hunk 2]
```diff
@@ -43,8 +46,21 @@ class SiteTitleControl extends React.Component {
 		this.props.onChange( { blogname, blogdescription } );
 	};
 
+	renderValidation = blogname => {
+		const isError = blogname ? false : true;
```
Comment (ockham) [inline client/components/site-title/index.jsx]: Ternary yielding a bool? There's got to be an easier way 😉

File: client/components/site-title/index.jsx [hunk 3]
```diff
@@ -43,8 +46,21 @@ class SiteTitleControl extends React.Component {
...
+
+		return <FormInputValidation isError={ isError } text="Required field." />;
```
Comment (ockham) [inline client/components/site-title/index.jsx]: The `text` attr needs `translate()`ing :slightly_smiling_face:

File: client/components/site-title/index.jsx [hunk 4]
```diff
@@ -54,9 +70,9 @@ class SiteTitleControl extends React.Component {
 						disabled={ disabled }
 						id="blogname"
 						onChange={ this.onChangeSiteTitle }
-						required
 						value={ blogname }
 					/>
+					{ isTitleRequired ? this.renderValidation( blogname ) : null }
```
Comment (ockham) [inline client/components/site-title/index.jsx]: I generally try to avoid `this.render...()` methods -- I find them sometimes distracting since they tempt me to scroll around, losing focus more easily.

I think we might be able to simply inline `<FormInputValidation isError={ isBlognameRequired && ! blogname } text={ translate( 'Required field.' ) />` here -- IMO that also reads self-explanatory enough, what do you think?

Comment (ockham) [reply client/components/site-title/index.jsx]: (Might need to still wrap with `{ isTitleRequired && ... }`)

File: client/jetpack-onboarding/steps/site-title.jsx [hunk 5]
```diff
@@ -78,10 +78,15 @@ class JetpackOnboardingSiteTitleStep extends React.PureComponent {
 							blogname={ this.state.blogname }
 							blogdescription={ this.state.blogdescription }
 							disabled={ isRequestingSettings }
+							isTitleRequired={ true }
```
Comment (ockham) [inline client/jetpack-onboarding/steps/site-title.jsx]: Let's shorten this to just `isTitleRequired` :slightly_smiling_face: JSX best practice FTW!

File: client/components/site-title/index.jsx [hunk 6]
```diff
@@ -43,8 +46,22 @@ class SiteTitleControl extends React.Component {
 		this.props.onChange( { blogname, blogdescription } );
 	};
 
+	renderValidation = blogname => {
+		const { translate } = this.props;
+		const isError = ! blogname;
+
+		return <FormInputValidation isError={ isError } text={ translate( 'Required field.' ) } />;
```
Comment (a8ci18n) [inline client/components/site-title/index.jsx]: Hi! I've found a possible matching string that has already been translated **40** times:
`translate( 'Required Fields' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by AnnaMag** [comment]
Feedback addressed 👍

File: client/components/site-title/index.jsx [hunk 7]
```diff
@@ -54,9 +64,11 @@ class SiteTitleControl extends React.Component {
 						disabled={ disabled }
 						id="blogname"
 						onChange={ this.onChangeSiteTitle }
-						required
 						value={ blogname }
 					/>
+					{ isBlognameRequired && (
+						<FormInputValidation isError={ ! blogname } text={ translate( 'Required field.' ) } />
```
Comment (a8ci18n) [inline client/components/site-title/index.jsx]: Hi! I've found a possible matching string that has already been translated **40** times:
`translate( 'Required Fields' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by tyxla** [comment]
A small comment here if I may: it feels a little odd to me to display a "✅ success" message when everything is fine with a field. Instead I'd assume that we display a form input validation message only when there is an error; and if everything is good, to NOT display anything. For example, see how the WP.com login form works.
